diff -Naur coreutils-8.28/lib/fts.c coreutils-8.28.tpg/lib/fts.c
--- coreutils-8.28/lib/fts.c	2017-09-01 07:12:43.000000000 +0000
+++ coreutils-8.28.tpg/lib/fts.c	2017-09-07 22:40:29.689637865 +0000
@@ -202,7 +202,7 @@
 # if __GNUC__ < 7
 #  define FALLTHROUGH ((void) 0)
 # else
-#  define FALLTHROUGH __attribute__ ((__fallthrough__))
+#  define FALLTHROUGH [[clang::fallthrough]]
 # endif
 #endif
 
diff -Naur coreutils-8.28/lib/xalloc-oversized.h coreutils-8.28.tpg/lib/xalloc-oversized.h
--- coreutils-8.28/lib/xalloc-oversized.h	2017-09-01 07:12:43.000000000 +0000
+++ coreutils-8.28.tpg/lib/xalloc-oversized.h	2017-09-07 22:35:24.815626247 +0000
@@ -43,7 +43,7 @@
 
 #if 7 <= __GNUC__
 # define xalloc_oversized(n, s) \
-   __builtin_mul_overflow_p (n, s, (__xalloc_count_type) 1)
+    ({ __xalloc_count_type __xalloc_count; __builtin_mul_overflow (n, s, &__xalloc_count); })
 #elif 5 <= __GNUC__ && !defined __ICC && !__STRICT_ANSI__
 # define xalloc_oversized(n, s) \
    (__builtin_constant_p (n) && __builtin_constant_p (s) \
