diff -up coreutils-9.4/lib/posixtm.c.omv~ coreutils-9.4/lib/posixtm.c
--- coreutils-9.4/lib/posixtm.c.omv~	2024-02-09 02:10:43.154032146 +0100
+++ coreutils-9.4/lib/posixtm.c	2024-02-09 02:13:19.071772954 +0100
@@ -191,7 +191,7 @@ posixtime (time_t *p, const char *s, uns
              | (tm0.tm_min ^ tm1.tm_min)
              | (tm0.tm_sec ^ tm1.tm_sec)))
         {
-          if (ckd_add (&t, t, leapsec))
+          if (ckd_add (&t, t, leapsec ? 1 : 0))
             return false;
           *p = t;
           return true;
