--- coreutils-8.22/lib/mgetgroups.c.hdrs~	2013-12-04 15:53:33.000000000 +0100
+++ coreutils-8.22/lib/mgetgroups.c	2014-05-23 19:21:43.031666022 +0200
@@ -32,6 +32,7 @@
 
 #include "getugroups.h"
 #include "xalloc-oversized.h"
+#include "xalloc.h"
 
 static gid_t *
 realloc_groupbuf (gid_t *g, size_t num)
--- coreutils-8.22/src/expand.c.hdrs~	2014-05-23 19:21:43.031666022 +0200
+++ coreutils-8.22/src/expand.c	2014-05-23 19:22:18.126678936 +0200
@@ -41,6 +41,7 @@
 /* Get mbstate_t, mbrtowc(), wcwidth(). */
 #if HAVE_WCHAR_H
 # include <wchar.h>
+# include <wctype.h>
 #endif
 
 /* Get iswblank(). */
