diff -Naur coreutils-8.24/lib/mgetgroups.c coreutils-8.24.tpg/lib/mgetgroups.c
--- coreutils-8.24/lib/mgetgroups.c	2015-07-01 15:08:30.000000000 +0000
+++ coreutils-8.24.tpg/lib/mgetgroups.c	2015-07-14 20:30:14.359563267 +0000
@@ -32,6 +32,7 @@
 
 #include "getugroups.h"
 #include "xalloc-oversized.h"
+#include "xalloc.h"
 
 /* Work around an incompatibility of OS X 10.11: getgrouplist
    accepts int *, not gid_t *, and int and gid_t differ in sign.  */
