--- coreutils-8.19/src/stty.c.tcsadrain~	2012-07-21 16:54:31.000000000 +0200
+++ coreutils-8.19/src/stty.c	2012-08-21 05:46:40.352604844 +0200
@@ -1005,7 +1005,7 @@ main (int argc, char **argv)
          spurious difference in an uninitialized portion of the structure.  */
       static struct termios new_mode;
 
-      if (tcsetattr (STDIN_FILENO, TCSADRAIN, &mode))
+      if (tcsetattr (STDIN_FILENO, TCSANOW, &mode))
         error (EXIT_FAILURE, errno, "%s", device_name);
 
       /* POSIX (according to Zlotnick's book) tcsetattr returns zero if
