--- coreutils-8.21/lib/mgetgroups.c~	2013-02-19 06:54:49.715373020 +0100
+++ coreutils-8.21/lib/mgetgroups.c	2013-02-19 06:54:50.966388024 +0100
@@ -32,6 +32,7 @@
 
 #include "getugroups.h"
 #include "xalloc-oversized.h"
+#include "xalloc.h"
 
 static gid_t *
 realloc_groupbuf (gid_t *g, size_t num)
--- coreutils-8.21/src/expand.c~	2013-02-19 06:52:47.150984408 +0100
+++ coreutils-8.21/src/expand.c	2013-02-19 06:52:48.296994693 +0100
@@ -41,6 +41,7 @@
 /* Get mbstate_t, mbrtowc(), wcwidth(). */
 #if HAVE_WCHAR_H
 # include <wchar.h>
+# include <wctype.h>
 #endif
 
 #include "system.h"
